Orange/data/io.py

```diff
@@ -352,9 +373,12 @@ def _equal_length(lst):
             flag = Flags(Flags.split(flags[col]))
             if flag.i: continue
 
+            type_flag = types and types[col].strip()
             try:
-                orig_values = [np.nan if i.strip() in MISSING_VALUES else i.strip()
-                               for i in data[:, col]]
+                orig_values = [np.nan if i.strip() in MISSING_VALUES and
+                                         type_flag not in
+                                         StringVariable.TYPE_HEADERS
```

Comment:
How about avoiding evaluating this condition for each value. Python doesn't (as of yet) simplify comprehensions even when the condition is known to be static.

More importantly, what if `type_flag` is unspecified and the values will only later in the lines below be found to, in fact, constitute a `StringVariable`?

It might be better to replace `nan` values there and then.